### PR TITLE
Avoid sending a chatmessage that a player joined if in singleplayer mode

### PR DESCRIPTION
**Pull request by red-001**

### CODE REVIEW

**Comment by t4im** [comment]
Why even register the callbacks in that case? :)
It's not like the singleplayer state suddenly changes after init.

File: builtin/game/misc.lua [hunk 1]
```diff
@@ -87,12 +87,14 @@ end
 local player_list = {}
 
 core.register_on_joinplayer(function(player)
+	if minetest.is_singleplayer() then return end
 	local player_name = player:get_player_name()
 	player_list[player_name] = player
```
Comment (t4im) [inline builtin/game/misc.lua]: Actually, are you sure the player_list isn't needed? `get_connected_players` might be used by mods and likely needs the user to be in that list, even in singleplayer.

File: builtin/game/misc.lua [hunk 2]
```diff
@@ -87,12 +87,14 @@ end
 local player_list = {}
 
 core.register_on_joinplayer(function(player)
+	if minetest.is_singleplayer() then return end
```
Comment (rubenwardy) [inline builtin/game/misc.lua]: I'd prefer:

```lua
if minetest.is_singleplayer() then
 	return end
end
```

File: builtin/game/misc.lua (hunk 1 above)
Comment (rubenwardy) [inline builtin/game/misc.lua]: :+1:

**Comment by paramat** [comment]
:-1: Singleplayer is a server with a player joining, so it's correct and consistent to show this message, it certainly isn't a problem so why make things more complex?

File: builtin/game/misc.lua (hunk 2 above)
Comment (paramat) [inline builtin/game/misc.lua]: Putting this in one line is actually strictly disallowed.

**Comment by SmallJoker** [comment]
The current behaviour makes it easier for newbies to see their player name, in case they wonder.

File: builtin/game/misc.lua [hunk 3]
```diff
@@ -87,14 +87,20 @@ end
 local player_list = {}
 
 core.register_on_joinplayer(function(player)
-	local player_name = player:get_player_name()
 	player_list[player_name] = player
```
Comment (sfan5) [inline builtin/game/misc.lua]: `player_name` is undefined here

**Comment by est31** [comment]
:+1: if the comment by @sfan5 is fixed.

**Comment by kwolekr** [comment]
Can we get a full tally of votes for and against the concept of this change?

Code review-wise it's a :+1:.  But as for feature concept review, we need a better consensus from players and other developers.

**Comment by bendeutsch** [comment]
I don't think non-technical players will understand that single-player and multi-player are the same, or that this "new player" joining the game is them. And if you need to know your name, you're already doing something technical, right? 

How about this: instead of suppressing it, change it to "Welcome, singleplayer" or something?

I personally don't mind either way, but then I consider myself a technical player.

**Comment by HybridDog** [comment]
Those non-technical players play a bit minetest without mods and then uninstall it and tell it's a crappy game, the other one is a lot better.

**Comment by everamzah** [comment]
Yet another differing change between so-called "singleplayer" mode and, what I like to refer to as, "normal mode."  :-1:

**Comment by est31** [comment]
I'm neutral now concept wise.

**Comment by rubenwardy** [comment]
I feel like it's more consistent to show the same join message as when a player joins a server.
Is this more confusing for players?

**Comment by PureTryOut** [comment]
Seeing as you can grant yourself permissions using `/grantme <permissionname|all>` (you don't have to use a username anymore), I see no reason for having a join message other than being consistent. However I think that showing such a message might be confusing for non-technical players. It seems to be a really small change anyway, so :+1: from me.

**Comment by C1ffisme** [comment]
I'm not sure why this is controversial, just because "Some newbie won't know what their name is in Single Player!"

I don't really see why this is important any more, since as @PureTryOut said, you don't need to use your username in many commands any more. I also don't see why we need to be consistent with Multiplayer. If anything, Single Player and Multiplayer should be more different.

**Comment by XtremeHacker** [comment]
I'll chime in as a non-newbie player.
Even though singleplayer is basically something like starting ther server in the background, and connecting the client via `localhost` , It's still singleplayer, and you can't have another person connect when you are playing singleplayer (correct me if I'm wrong), so having the 

> UsernameHere joined the game
Makes no sense IMHO, and the code isn't sloppy, so basically, why not merge?
This is a game after all, and sure most people who play it (myself included) can undertsand computers more then the average Joe/Jane, the average Joe/Jane are a part of the intended audience, so once again, why not merge?

**Comment by Fixer-007** [comment]
Thumb up or down original message.

**Comment by paramat** [comment]
Is code change needed in 'on leaveplayer'? Does the message pop up in chat just before singleplayer leaves?

File: builtin/game/misc.lua [hunk 4]
```diff
@@ -89,12 +89,18 @@ local player_list = {}
 core.register_on_joinplayer(function(player)
 	local player_name = player:get_player_name()
 	player_list[player_name] = player
+	if minetest.is_singleplayer() then
```
Comment (rubenwardy) [inline builtin/game/misc.lua]: This early exit isn't required and is ugly imo. I'd prefer this:

```lua
if not minetest.is_singleplayer() then
    core.chat_send_all("*** " .. player_name .. " joined the game.")
end
```

Comment (paramat) [inline builtin/game/misc.lua]: Agreed.

**Comment by paramat** [comment]
Seems good now.

**Comment by rubenwardy** [comment]
:+1:

**Comment by sfan5** [comment]
fb1406bdff6b906ae5a04c34fe58cc57aae6d31e

**Pull request closed without merging**